core/domain/story_domain.py

```diff
@@ -0,0 +1,369 @@
+        """Constructs a Question domain object.
+
+        Args:
+            story_id: str. The unique ID of the story.
+            title: str. The title of the story.
+            description: str. The high level desscription of the story.
+            notes: str. A set of notes, that describe the characters,
+                main storyline, and setting.
+            story_nodes: list(StoryNode). List of the story nodes present in
+                the story.
+            created_on: datetime.datetime. Date and time when the story is
+                created.
+            last_updated: datetime.datetime. Date and time when the
+                story was last updated.
+            schema_version: int. The schema version for the story nodes object.
+            language_code: str. The ISO 639-1 code for the language this
+                story is written in.
+            version: int. The version of the story.
+        """
+        self.id = story_id
+        self.title = title
+        self.description = description
+        self.notes = notes
+        self.story_nodes = story_nodes
+        self.schema_version = schema_version
+        self.language_code = language_code
+        self.created_on = created_on
+        self.last_updated = last_updated
+        self.version = version
+
+    def to_dict(self):
+        """Returns a dict representing this Story domain object.
+
+        Returns:
+            A dict, mapping all fields of Story instance.
+        """
+        return {
+            'id': self.id,
+            'title': self.title,
+            'description': self.description,
+            'notes': self.notes,
+            'language_code': self.language_code,
+            'schema_version': self.schema_version,
+            'version': self.version,
+            'story_nodes': [
+                node.to_dict() for node in self.story_nodes
+            ]
+        }
+
+    @classmethod
+    def from_dict(
+            cls, story_dict,
+            story_created_on=None, story_last_updated=None):
+        """Return a Story domain object from a dict.
+
+        Args:
+            story_dict: dict. The dictionary representation of the
+                story.
+            story_created_on: datetime.datetime. Date and time when the
+                story is created.
+            story_last_updated: datetime.datetime. Date and time when
+                the story was last updated.
+
+        Returns:
+            Story. The corresponding Story domain object.
+        """
+        story = cls(
+            story_dict['id'], story_dict['title'],
+            story_dict['description'], story_dict['notes'],
+            [
+                StoryNode.from_dict(node_dict)
+                for node_dict in story_dict['story_nodes']
+            ], story_dict['schema_version'], story_dict['language_code'],
+            story_dict['version'], story_created_on, story_last_updated)
+
+        return story
+
+    @classmethod
+    def create_default_story(
```

Comment:
Is this for the frontend? If so then just have the args be (cls, story_id) and explain in the docstring that this is only to be used for initializing a brand-new story. If not then what else is this intended to be used for?

Reply:
Yeah, this is for the frontend.